packages/apptive_grid_heinzelmen/lib/src/invitation_dialog/invitation_dialog.dart

```diff
@@ -0,0 +1,344 @@
+  required BuildContext context,
+  required Space space,
+  List<Role> allowedRoles = Role.values,
+  ApptiveLinkType type = ApptiveLinkType.invite,
+  InvitationLocalization? localization,
+}) async {
+  assert(type == ApptiveLinkType.invite || type == ApptiveLinkType.addShare);
+  assert(space.links.containsKey(type));
+  assert(allowedRoles.isNotEmpty);
+
+  final l10n = localization ??
+      (Localizations.localeOf(context).languageCode == 'de'
+          ? const InvitationLocalizationDE()
+          : const InvitationLocalizationEn());
+
+  TextEditingController emailController = TextEditingController();
+  ValueNotifier<Role> roleNotifier = ValueNotifier(allowedRoles.first);
+  ValueNotifier<bool> loading = ValueNotifier(false);
+  ValueNotifier<dynamic> errorNotifier = ValueNotifier(null);
+  bool dialogResult = false;
+  return showDialog(
+    context: context,
+    barrierDismissible: false,
+    builder: (dialogContext) {
+      return ValueListenableBuilder<bool>(
+        valueListenable: loading,
+        builder: (_, loading, child) => IgnorePointer(
+          ignoring: loading,
+          child: child,
+        ),
+        child: Form(
+          child: AlertDialog(
+            title: Text.rich(
+              TextSpan(
+                children: getSpans(
+                  getText: l10n.title,
+                  argument: space.name,
+                  style: TextStyle(
+                    color: Theme.of(context).colorScheme.primary,
+                  ),
+                ),
+              ),
+            ),
+            content: _InvitationDialogContent(
+              emailController: emailController,
+              role: roleNotifier,
+              error: errorNotifier,
+              allowedRoles: allowedRoles,
+              localization: l10n,
+            ),
+            actions: [
+              TextButton(
+                onPressed: Navigator.of(dialogContext).pop,
+                child: Text(l10n.actionCancel),
+              ),
+              _InvitationSendButton(
+                loading: loading,
+                role: roleNotifier,
+                error: errorNotifier,
+                onInviteSend: (value) => dialogResult = value,
+                localization: l10n,
+                email: emailController,
+                link: space.links[type]!,
+              ),
+            ],
+          ),
+        ),
+      );
+    },
+  ).then((_) => dialogResult);
+}
+
+/// Returns a List of [TextSpans] that format [argument] with [style] and use default styles for other text that was generated with [getText]
+@visibleForTesting
+List<TextSpan> getSpans({
+  required String Function(String) getText,
+  required String argument,
+  required TextStyle style,
+}) {
+  final emptyTitle = getText('');
+  final realTitle = getText(argument);
+
+  if (realTitle.length == emptyTitle.length) {
+    return [TextSpan(text: realTitle)];
+  } else {
+    int startIndex = emptyTitle.length;
```

Comment:
If we support custom localizations, I guess we also need to handle multiple occurrences of the space name.

Reply:
Theoretically yes.
However I am not sure if the added logic really makes that much sense to add right now. As this method is not public to the outside (just visible for testing).
I struggle to find sensible ways of what texts would be that would require the space name twice or the email twice. I think if we or someone using this it should be solved then.

Reply:
Fine by me